src/winter_onboarding_2021/4clojure/shivam.clj

```diff
@@ -0,0 +1,276 @@
+
+;; (defn less-mass-than-earth?
+;;   [planet]
+;;   (< (:mass planet) 1))
+
+
+
+
+;; (def planetsSmallerThanEarth (map :mass (filter less-mass-than-earth? planets))) ;; (0.52 0.8)
+
+;; (reduce + 0 planetsSmallerThanEarth)
+
+
+
+
+;; EXERCISE
+;; Define a predicate `poison-gas?` which returns the
+;; poison gas if it belongs to a set of known poison gases,
+;; or `nil` (falsey) otherwise. These are some known poison gases:
+
+
+;; (defn poison-gas?
+;;   [gas]
+;;   ;"Does the given gas belong to a set of known poison gases?"
+;;   (boolean (#{:carbon-monoxide, :chlorine, :helium
+;;     :sulphur-dioxide, :hydrogen-chloride} gas))
+;;   )
+
+
+;; (poison-gas? :chlorine)                 ; truthy
+;; (poison-gas? :oxygen)                   ; falsey
+
+; I am loving clojure
+
+
+
+
+
+;; (map (fn [f] (f 42))
+;;      [str identity inc dec (fn [x] x)])
+
+;; ;; A data table:
+;; [[:name :age :country]
+;;  ["Foo" 10   "India"]
+;;  ["Bar" 21   "Australia"]
+;;  ["Baz" 18   "Turkey"]
+;;  ["Qux" 42   "Chile"]]
+
+
+;; (= '(1 2 3 4) (conj '(2 3 4) 1))
+
+(defn problem5 [] '(1 2 3 4))
+
+(defn problem6 [] [:a :b :c])
+
+(defn problem7 [] [1 2 3 4])
+
+(defn problem8 [] #{:a :b :c :d})
+
+(defn problem9 [] 2)
+
+(defn problem10 [] 20)
+
+(defn problem11 [] {:b 2})
+
+(defn problem12 [] 3)
+
+(defn problem13 [] '(20 30 40))
+
+(defn problem14 [] 8)
+
+(defn problem15 [x] (* x 2))
+
+(defn problem16 [name] (str "Hello, " name "!"))
+
+(defn problem17 [] '(6 7 8))
+
+(defn problem18 [] '(6 7))
+
+(defn problem19 [] last)
+
+(defn problem20 [coll] (second (reverse coll)))
+
+(defn problem21 [coll index] (if (>= index 0)
+                               ((vec coll) index)
+                               ((vec coll) (+ (count coll) index))))
+
+(defn problem22 [seq-coll] (reduce (fn [acc _] (+ acc 1)) 0 seq-coll))
+
+(defn problem23 [coll] (reduce
+                        (fn [acc, ele] (cons ele acc))
+                        (empty coll)
+                        coll))
+
+(defn problem24 [coll] (reduce
+                        (fn [acc ele] (+ acc ele))
+                        0
+                        coll))
+
+(defn problem25 [coll] (filter
+                        (fn [x] (not (= 0 (rem x 2))))
+                        coll))
+
+(defn problem26 [n] (reduce
+                     (fn [acc, ele]
+                       (if (<= n 1)
+                         (vec (range n))
+                         (conj acc (+ (or (second (reverse acc)) 1) (last acc)))))
+                     (if (<= n 1)
+                       []
+                       [0])
+                     (range (- n 1))))
+
+(defn problem27 [coll] (if (= java.lang.String (type coll))
+                         (= (apply str (reverse coll)) coll)
+                         (= (reverse coll) coll)))
+
+
+(defn problem28 [coll] (reduce
+                        (fn [acc, ele] (concat acc (if (coll? ele)
+                                                     (problem28 ele)
```

Comment:
You are calling the same function inside of reduce. This looks like a tell-tell sign that recursion would work better here. 
See you can rewrite the solution using recursion.